src/main/webapp/script.js

```diff
@@ -72,10 +72,18 @@ function postData(url, data) {
       .catch((error) => handleFetchError(error));
 }
 
+function handleNetworkError(exception){
+  console.error("There was a network error that prevented the completition of the request.");
+  console.error(exception);
+}
+
 function getJsonIfOk(response) {
   if (!response.ok) {
-    console.error('Network error detected');
+    console.error('Server error detected: ' + respons.status);
+    console.error(response.statusText);
+    throw response.status;
```

Comment:
I recommend always throwing an Error instead of e.g. a number like here. It's often not clear how this has to be interpreted and most people are not aware how this is wrapped in a exception automatically.
See:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw